File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<int> freq(2001, 0);
        vector<int> num;
        for(auto n:nums) {
            freq[n + 1000]++;
            if (freq[n + 1000] == 1) num.push_back(n);
        }
        sort(num.begin(), num.end(), [&](int a, int b) {
            return freq[a + 1000] > freq[b + 1000];
        });

        return vector<int>(num.begin(), num.begin() + k);
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dt = {}
        n = len(nums)
        for i in range(n):
            num = nums[i]
            if (target - num) in dt:
                return [dt[target - num], i]
            dt[num] = i
        return [-1, -1]
